### PR TITLE
PHOENIX-4864 Fix NullPointerException while Logging some DDL Statements

### PR DESCRIPTION
**Pull request by ashuparekh**
Fixed the error. Still to add corresponding test case. @vincentpoon @karanmehta93

### CODE REVIEW

File: phoenix-core/src/main/java/org/apache/phoenix/jdbc/LoggingPhoenixPreparedStatement.java [hunk 1]
```diff
@@ -45,7 +45,9 @@ public ResultSet executeQuery() throws SQLException {
 
     @Override
     public ResultSet getResultSet() throws SQLException {
-        return new LoggingPhoenixResultSet(super.getResultSet(), phoenixMetricsLog, sql);
+        ResultSet resultSet = super.getResultSet();
+        return (resultSet == null) ? null : new LoggingPhoenixResultSet(super.getResultSet(),
```
Comment (karanmehta93) [inline phoenix-core/src/main/java/org/apache/phoenix/jdbc/LoggingPhoenixPreparedStatement.java]: This is still a bug. `super.getResultSet()` call is not idempotent. You should replace this with `resultSet` local variable

**Comment by ashuparekh** [comment]
Added the corresponding test case. @vincentpoon @karanmehta93

File: phoenix-core/src/it/java/org/apache/phoenix/monitoring/PhoenixLoggingMetricsIT.java [hunk 2]
```diff
@@ -91,19 +125,19 @@ public void testPhoenixMetricsLoggedOnCommit() throws Exception {
 
         // Assert that metrics logging happens only once
         loggedConn.close();
-        assertTrue("Mutation write metrics are not logged again.",
-                mutationWriteMetricsMap.size() == 0);
-        assertTrue("Mutation read metrics are not logged again.",
-                mutationReadMetricsMap.size() == 0);
+        assertTrue("Mutation write metrics are not logged again.", mutationWriteMetricsMap.size()
```
Comment (karanmehta93) [inline phoenix-core/src/it/java/org/apache/phoenix/monitoring/PhoenixLoggingMetricsIT.java]: nit: unnecessary diff (in lot of places)

File: phoenix-core/src/it/java/org/apache/phoenix/monitoring/PhoenixLoggingMetricsIT.java [hunk 3]
```diff
@@ -32,6 +35,7 @@
 
 import static org.junit.Assert.assertTrue;
 
+
```
Comment (karanmehta93) [inline phoenix-core/src/it/java/org/apache/phoenix/monitoring/PhoenixLoggingMetricsIT.java]: nit

File: phoenix-core/src/main/java/org/apache/phoenix/jdbc/LoggingPhoenixPreparedStatement.java [hunk 4]
```diff
@@ -45,7 +45,9 @@ public ResultSet executeQuery() throws SQLException {
 
     @Override
     public ResultSet getResultSet() throws SQLException {
-        return new LoggingPhoenixResultSet(super.getResultSet(), phoenixMetricsLog, sql);
+        ResultSet resultSet = super.getResultSet();
```
Comment (karanmehta93) [inline phoenix-core/src/main/java/org/apache/phoenix/jdbc/LoggingPhoenixPreparedStatement.java]: nit: add a comment that call to `getResultSet()` is not idempotent. Hence we cache and return accordingly.

**Comment by karanmehta93** [comment]
Nice test coverage @ashuparekh 
This PR is almost good to go!

**Comment by ashuparekh** [comment]
@karanmehta93 Incorporated the above suggestions

**Comment by stoty** [comment]
Already merged.
